src/Arcus.WebApi.Security/Authorization/Jwt/JwtTokenReader.cs

```diff
@@ -63,29 +72,54 @@ public JwtTokenReader(TokenValidationParameters tokenValidationParameters, strin
         /// <param name="token">The JWT token.</param>
         public async Task<bool> IsValidTokenAsync(string token)
         {
+            bool result;
+
             try
             {
-                TokenValidationParameters validationParameters = await DetermineTokenValidationParametersAsync();
-                _handler.ValidateToken(token, validationParameters, out SecurityToken jwtToken);
+                var validationParameters = await DetermineTokenValidationParametersAsync();
+                _handler.ValidateToken(token, validationParameters, out var validatedToken);
+                result = ValidateClaimCheck(validatedToken);
             }
-            catch (Exception exception)
+            catch
             {
                 return false;
             }
 
-            return true;
+            return result;
+        }
+
+        /// <summary>
+        /// Verifies if the claims coming in ClaimCheck object vs the claims in the jwt object based on the claim name and value are valid
+        /// </summary>
+        /// <param name="validatedToken">Security token which contains the jwt claims</param>
+        /// <returns></returns>
+        public bool ValidateClaimCheck(SecurityToken validatedToken)
+        {
+            if (_claimCheck == null || !_claimCheck.Any())
+            {
+                return true;
+            }
+
+            var jwtToken = (JwtSecurityToken) validatedToken;
+
+            var validClaims = (from c in jwtToken.Claims
+                join m in _claimCheck on c.Type equals m.Key
+                where c.Value == m.Value
+                select c).Distinct().Count();
+
+            return _claimCheck.Count == validClaims;
         }
 
         private async Task<TokenValidationParameters> DetermineTokenValidationParametersAsync()
         {
             if (_tokenValidationParameters is null)
             {
-                OpenIdConnectConfiguration config = await _configManager.GetConfigurationAsync();
-                
+                var config = await _configManager.GetConfigurationAsync();
+
                 var validationParameters = new TokenValidationParameters
                 {
-                    ValidateAudience = false,
-                    ValidAudience = _applicationId,
+                    ValidateAudience = _claimCheck.ContainsKey(AudienceKeyName),
+                    ValidAudience = _claimCheck[AudienceKeyName],
```

Comment:
This will throw an exception if AudienceKeyName is not present in the _claimCheck dictionary.

(Would be good to create a testcase for this)

Reply:
Yes, a `Guard` would be sufficient here.

Reply:
A Guard will not solve the issue.  
The fact is , that checking the audience is optional, and should only happen when there's a claimcheck defined for this.

Code should be something like this:

```csharp
ValidateAudience = _claimChecks.ContainsKey(AudienceKeyName);
ValidAudience = (_claimChecks.ContainsKey(AudienceKeyName) ? _claimChecks[AudienceKeyName] : null
```

Reply:
done